lib/Cron/Supervisor.php

```diff
@@ -6,34 +6,41 @@ class Supervisor {
   public $daemon;
   public $token;
 
-  function __construct() {
-    $this->token = CronHelper::createToken();
+  /** @var CronHelper */
+  private $cron_helper;
+
+  function __construct(CronHelper $cron_helper) {
+    $this->cron_helper = $cron_helper;
+  }
+
+  function init() {
```

Comment:
This can make the service "invalid" when injected somewhere if `init` wasn't called. I have an idea:
- Instead of `$this->daemon` use `$this->getDaemon()` inside this class - it already kind of behaves as a lazy property, maybe `$this->daemon` is not needed at all. Or... it can be lazy.
- Make token lazy as well (with private getter method), if needed (= if you think it shouldn't be generated in constructor).

Reply:
I've introduced `init` because there was a problem with `Triggers` tests sharing the state of this class. I'll just use `setShared(false)` in the DI container and restore the original constructor.